src/main/java/com/google/devtools/build/lib/remote/RemoteModule.java

```diff
@@ -135,7 +145,9 @@ private void verifyServerCapabilities(
       CallCredentials credentials,
       RemoteRetrier retrier,
       CommandEnvironment env,
-      DigestUtil digestUtil)
+      DigestUtil digestUtil,
+      boolean shouldCheckExecutionCapabilities,
+      boolean shouldCheckCacheCapabilities)
```

Comment:
What about replacing this with an enum:

```java
private enum Capabilities {CACHE_ONLY, EXECUTION_ONLY, CACHE_AND_EXECUTION }
```

Then the code below with the various true/false permutations to `verifyServerCapabilities` becomes easier to read.